pkg/network/ebpf/c/protocols/postgres/decoding.h

```diff
@@ -180,12 +194,72 @@ static __always_inline void postgres_handle_parse_message(pktbuf_t pkt, conn_tup
     return;
 }
 
+// Handles Postgres command complete messages by examining packet data for both plaintext and TLS traffic.
+// This function handles multiple messages within a single packet, processing up to POSTGRES_MAX_MESSAGES_PER_TAIL_CALL
+// messages per call. When more messages exist beyond this limit, it uses tail call chaining (up to
+// POSTGRES_MAX_TAIL_CALLS_FOR_MAX_MESSAGES) to continue processing.
+static __always_inline bool handle_command_complete_messages(pktbuf_t pkt, conn_tuple_t conn_tuple) {
+    const __u32 zero = 0;
+    struct pg_message_header header;
+    if (!read_message_header(pkt, &header)) {
+        return 0;
+    }
```

Comment:
read message header only when you're required.
Since we already read the message header in the parent tail call, we have high confidence it will succeed. So let's "waste" the resources only after we know the other checks succeeded

Reply:
actually, seems like a redundant operation.

Reply:
you are right also :)